src/cdk/drag-drop/drag-ref.ts

```diff
@@ -977,6 +978,16 @@ export class DragRef<T = any> {
     this._pointerPositionAtLastDirectionChange = {x: pointerPosition.x, y: pointerPosition.y};
     this._dragStartTime = Date.now();
     this._dragDropRegistry.startDragging(this, event);
+
+    // when pixel threshold = 0 and dragStartDelay = 0 and a preview container/position exists we immediately drag
+    if (
+      (event.type == 'mousedown' || event.type == 'touchstart') &&
```

Comment:
I'm not sure I follow this. So the only behavior difference here is that the preview will show up on the `mousedown` rather than the next `mousemove`?

Reply:
See this chessboard for example: 

[Chess.com Explorer](https://www.chess.com/explorer)

Click on any piece to move it and you will see the drag immediately starts. This is the effect of the snapToCursor combined with immediate preview rendering (on mousedown/touchstart).

Reply:
I think in that case a better way to approach it would be to check that threshold/delay are at 0 and then call `_startDragSequence` from the "down" handler. It might be necessary to move some logic out of `_pointerMove` so it can be reused.

Reply:
@crisbeto I had looked at that -- however the entirety of the `_pointerMove` function must be executed in this case especially around the constrainPosition and other logic executed in that method in addition to `_startDragSequence` as a setup step being called.

I had tried to limit it to just a few calls but that either breaks functionality or results in excess redundant code where this version cleanly was backwards-compatible.

Reply:
My thinking was that we can move the whole block that's guarded by `!this._hasStartedDragging()` out into a separate function. The rest shouldn't be executed before dragging has started anyways.

Reply:
@chrisbeto unfortunately that doesn't work since the dragging must start when the mouse goes down...so the entirety of the pointerMove function is executed (note the return is skipped when it's a mouse down).